Revit_Core_Engine/Query/Shape.cs

```diff
@@ -68,6 +70,34 @@ public static Autodesk.Revit.DB.ConnectorProfileType Shape(this MEPCurve mEPCurv
         }
 
         /***************************************************/
+
+        [Description("Determines whether a MEP Family Symbol is round, rectangular or oval.")]
+        [Input("familySymbol", "Family Symbol to check in order to determine its shape.")]
+        [Input("doc", "Document object of the Family Symbol.")]
+        [Input("settings", "Revit adapter settings.")]
+        [Output("shape", "Shape of an Family Symbol, which can be round, rectangular or oval. If the shape cannot be determined, an invalid connector shape is returned.")]
+        public static ConnectorProfileType Shape(this FamilySymbol familySymbol, Document doc, RevitSettings settings = null)
+        {
+            if (familySymbol == null || doc == null)
+            {
+                BH.Engine.Base.Compute.RecordError("Family Symbol or input document cannot be null.");
+                return ConnectorProfileType.Invalid;
+            }
+
+            Document familyDoc = doc.EditFamily(familySymbol.Family);
+            List<ConnectorElement> connectors = new FilteredElementCollector(familyDoc).OfClass(typeof(ConnectorElement)).Cast<ConnectorElement>().ToList();
+
+            if (connectors.Count == 0)
+            {
+                BH.Engine.Base.Compute.RecordWarning("Family of the Family Symbol has no connectors, shape cannot be determined.");
+                return ConnectorProfileType.Invalid;
+            }
+
+            ConnectorProfileType shape = connectors.Where(x => x.IsPrimary).Select(x => x.Shape).FirstOrDefault();
+
+            return shape;
```

Comment:
I would consider shortening the method to avoid empty runs and also make the warning more consistent (to actually capture _only_ primary connectors). But please feel free to ignore if you dislike 😉 
```suggestion
            ConnectorProfileType? shape = new FilteredElementCollector(familyDoc).OfClass(typeof(ConnectorElement)).Cast<ConnectorElement>().FirstOrDefault(x => x.IsPrimary)?.Shape;

            if (shape == null)
            {
                BH.Engine.Base.Compute.RecordWarning("Family of the Family Symbol has no connectors, shape cannot be determined.");
                return ConnectorProfileType.Invalid;
            }
            
            return shape.Value;
```